computron: Update CMAKE_OSX_DEPLOYMENT_TARGET to 14.0 for preview builds

jamesobutler opened the pull request:
This PR aims to update the CMAKE_OSX_DEPLOYMENT_TARGET for Slicer on macOS to be a version of macOS that is still supported by Apple.

This will support the following macOS versions:
- 26.x Tahoe (2025)
- 15.x: Sequoia (2024)
- 14.x: Sonoma (2023)

This is a follow-up to https://github.com/Slicer/DashboardScripts/pull/73 and https://github.com/Slicer/DashboardScripts/commit/bd9d10a23c08becb7e7debff6b919df6a6d81143 where the macOS deployment target was updated from 11.0 to 13.0. This PR updates the deployment target to macOS 14 (Sonoma) now that macOS 26 (Tahoe) is out and Apple is no longer supporting macOS 13 (Ventura).

> [!NOTE] 
> The "computron" Slicer factory machine is running macOS 15.7.1 (Ventura) as seen in recent [discourse post](https://discourse.slicer.org/t/maintenance-of-macos-dashboard-in-progress-upgrading-from-ventura-to-sequoia/44950?u=jamesobutler).

jamesobutler committed:
computron: Update CMAKE_OSX_DEPLOYMENT_TARGET to 14.0 for preview builds

Although Apple does not list an end-of-support date, macOS Ventura (13.0) last received a security update in August 2025. https://endoflife.date/macos

sjh26 reviewed (commented):
We should update the script for the stable extension as well

sjh26 reviewed (approved):
lgtm

jamesobutler commented:
So pick off the latest commit `style: Consistently reference the same macOS deployment target`? Or only maintain "13.0" for computron-slicerextensions_stable_nightly.cmake?

jamesobutler committed:
style: Consistently reference the same macOS deployment target

This changes the macOS deployment target to "14.0" across various configuration files. While the variable is not used on Windows and Linux dashboards, this will avoid confusion, reduce the chance of copy-paste issues, and streamline file comparison.

computron-slicerextensions_stable_nightly.cmake is maintained as 13.0 because Slicer 5.10.0 stable was built targeting 13.0. computron-slicer_stable_package.cmake is maintain at 13.0 for consistency as well.
